src/librustc_mir/transform/const_prop.rs

```diff
@@ -707,11 +707,7 @@ impl<'mir, 'tcx> ConstPropagator<'mir, 'tcx> {
                 ScalarMaybeUninit::Scalar(l),
                 ScalarMaybeUninit::Scalar(r),
             )) => l.is_bits() && r.is_bits(),
-            interpret::Operand::Indirect(_) if mir_opt_level >= 2 => {
-                let mplace = op.assert_mem_place(&self.ecx);
-                intern_const_alloc_recursive(&mut self.ecx, InternKind::ConstProp, mplace, false);
```

Comment:
Why was this needed / why is this no longer needed? Seems like a surprising change to me.

Reply:
This was buggy code that basically wasn't doing anything before this PR and was causing crashes with this PR. It's a leftover that multiple refactorings made obsolete. I can remove it on master if you prefer.

Reply:
No it's fine. One less interning entry point to worry about. :)